File: Year_2/Semester_2/Medii_De_Proiectare_Si_Programare/FestivalTickets/Java_Gradle_Project/Laboratory_1/FestivalTicketSale/src/main/java/org/example/Model/Validators/ModelValidators/SpectacolValidator.java
package org.example.Model.Validators.ModelValidators;

import org.example.Model.Exceptions.ValidatorException;
import org.example.Model.Spectacol;
import org.example.Model.Validators.Interfaces.Validator;

public class SpectacolValidator implements Validator<Spectacol> {

    @Override
    public void validate(Spectacol entity) throws ValidatorException {
        String errors="";
        if(entity.getLocatie().isEmpty())
        {
            errors+="Locatia cannot be empty!\n";
        }
        if(entity.getNumar_locuri_disponibile()<0)
        {
            errors+="A negative number of available seats is not valid!\n";
        }
        if(entity.getNumar_locuri_vandute()<0)
        {
            errors+="A negative number of sold seats is not valid!\n";
        }
        if(entity.getArtist().isEmpty())
        {
            errors+="Artist cannot be empty!\n";
        }
        if(!errors.isEmpty())
        {
            throw new ValidatorException(errors);
        }
        }
}

File: Year_2/Semester_2/Medii_De_Proiectare_Si_Programare/FestivalTickets/C#_Project/Laboratory_8_Proto/FestivalTicketSaleCS/model/Validators/ModelValidators/Spectacol Validator.cs
using System;
using model.Exceptions;
using model.Validators.Interfaces;

namespace model.Validators.ModelValidators
{
    public class SpectacolValidator : IValidator<Spectacol>
    {
        public void Validate(Spectacol entity)
        {
            string errors = "";

            if (string.IsNullOrEmpty(entity.Locatie))
            {
                errors += "Locatia cannot be empty!\n";
            }
            if (entity.NumarLocuriDisponibile < 0)
            {
                errors += "A negative number of available seats is not valid!\n";
            }
            if (entity.NumarLocuriVandute < 0)
            {
                errors += "A negative number of sold seats is not valid!\n";
            }
            if (string.IsNullOrEmpty(entity.Artist))
            {
                errors += "Artist cannot be empty!\n";
            }
            if (!string.IsNullOrEmpty(errors))
            {
                throw new ValidatorException(errors);
            }
        }
    }
}